lib/active_data/model/associations/persistence_adapters/active_record.rb

```diff
@@ -0,0 +1,51 @@
+module ActiveData::Model::Associations::PersistenceAdapters
+  class ActiveRecord < Base
+    class ScopeProxy
+      # You can't create data directly through ActiveRecord::Relation
+      METHODS_EXCLUDED_FROM_DELEGATION = %w[build create create!].map(&:to_sym).freeze
+
+      attr_reader :original_scope
+
+      def initialize(original_scope)
+        @original_scope = original_scope
+      end
+
+      def method_missing(method, *args, &block)
+        delegate_to_original_scope?(method) ? original_scope.send(method, *args, &block) : super
+      end
+
+      def respond_to_missing?(method, include_private = false)
+        delegate_to_original_scope?(method) || super
+      end
+
+      def delegate_to_original_scope?(method)
+        METHODS_EXCLUDED_FROM_DELEGATION.exclude?(method) && original_scope.respond_to?(method)
+      end
+    end
+
+    def scope(source)
+      scope = (scope_proc ? klass.unscoped.instance_exec(&scope_proc) : klass.unscoped)
+      scope = scope.where(primary_key => source)
+      ScopeProxy.new(scope)
```

Comment:
Let's maybe initialize this in find_all? Also seems like find_all implementation screws the proxy up returning an array.